extensions/import/src/wizard/dacfxExportWizard.ts

```diff
@@ -0,0 +1,127 @@
+		let page1 = sqlops.window.modelviewdialog.createWizardPage(localize('dacFxExport.page1Name', 'Export Settings'));
+		let page2 = sqlops.window.modelviewdialog.createWizardPage(localize('dacFxExport.page2Name', 'Summary'));
+
+		page1.registerContent(async (view) => {
+			this.exportConfigPage = new ExportConfigPage(this, page1, this.model, view);
+			pages.set(0, this.exportConfigPage);
+			await this.exportConfigPage.start().then(() => {
+				this.exportConfigPage.setupNavigationValidator();
+				this.exportConfigPage.onPageEnter();
+			});
+		});
+
+		page2.registerContent(async (view) => {
+			this.summaryPage = new ExportSummaryPage(this, page2, this.model, view);
+			pages.set(1, this.summaryPage);
+			await this.summaryPage.start();
+		});
+
+		this.wizard.onPageChanged(async (event) => {
+			let idx = event.newPage;
+			let page = pages.get(idx);
+
+			if (page) {
+				page.setupNavigationValidator();
+				page.onPageEnter();
+			}
+		});
+
+		this.wizard.onPageChanged(async (event) => {
+			let idx = event.lastPage;
+			let page = pages.get(idx);
+
+			if (page) {
+				page.onPageLeave();
+			}
+		});
+
+		this.wizard.pages = [page1, page2];
+		this.wizard.generateScriptButton.hidden = true;
+		this.wizard.doneButton.label = localize('dacFxExport.exportButton', 'Export');
+		this.wizard.doneButton.onClick(async () => await this.export());
+		this.wizard.open();
+	}
+
+	private async export() {
+		let connectionString = await this.getConnectionString();
+		let packageFilePath = this.model.filePath;
+		let service = await DacFxExportWizard.getService();
+		let ownerUri = await sqlops.connection.getUriForConnection(this.model.serverConnection.connectionId);
+
+		let result = await service.exportBacpac(connectionString, packageFilePath, ownerUri, sqlops.TaskExecutionMode.execute);
+		if (!result || !result.success) {
+			vscode.window.showErrorMessage(
+				localize('alertData.saveErrorMessage', "Export failed '{0}'", result.errorMessage ? result.errorMessage : 'Unknown'));
+		}
+	}
+
+	public static async getService(): Promise<sqlops.DacFxServicesProvider> {
+		let currentConnection = await sqlops.connection.getCurrentConnection();
+		let service = sqlops.dataprotocol.getProvider<sqlops.DacFxServicesProvider>(currentConnection.providerName, sqlops.DataProviderType.DacFxServicesProvider);
+		return service;
+	}
+
+	public registerNavigationValidator(validator: (pageChangeInfo: sqlops.window.modelviewdialog.WizardPageChangeInfo) => boolean) {
+		this.wizard.registerNavigationValidator(validator);
+	}
+
+	private async getConnectionString(): Promise<string> {
+		let connectionString = await sqlops.connection.getConnectionString(this.model.serverConnection.connectionId, true);
+		let splitted = connectionString.split(';');
+
+		// set database to appropriate value instead of master
+		let temp = splitted.find(s => s.startsWith('Initial Catalog'));
```

Comment:
hmm, seems like using the connection management APIs would be a better way to go here.